reference/forms/types/choice.rst

```diff
@@ -85,19 +162,86 @@ choices
 
 This is the most basic way to specify the choices that should be used
 by this field. The ``choices`` option is an array, where the array key
-is the item value and the array value is the item's label::
+is the item's label and the array value is the item's value::
+
+    $builder->add('inStock', 'choice', array(
+        'choices' => array('In Stock' => true, 'Out of Stock' => false),
+        // always include this
+        'choices_as_values' => true,
+    ));
+
+choices_as_values
+~~~~~~~~~~~~~~~~~
+
+**type**: ``boolean`` **default**: false
+
+.. versionadded:: 2.7
+
+    The ``choices_as_values`` option was introduced in Symfony 2.7.
+
+The ``choices_as_values`` option was added to keep backward compatibility with the
+*old* way of handling the ``choices`` option. When set to ``false`` (or omitted),
+the choice keys are used as the underlying value and the choice values are shown
+to the user.
+
+* Before 2.7 (and deprecated now)::
 
     $builder->add('gender', 'choice', array(
-        'choices' => array('m' => 'Male', 'f' => 'Female'),
+        // Shows "Male" to the user, returns "m" when selected
+        'choices'  => array('m' => 'Male', 'f' => 'Female'),
+        'choices_as_values' => false,
     ));
 
-.. tip::
+* Optional since 2.7::
+
+    $builder->add('gender', 'choice', array(
+        // Shows "Male" to the user, returns "m" when selected
+        'choices' => array('Male' => 'm', 'Female' => 'f'),
+        'choices_as_values' => true,
+    ));
+
+In Symfony 3.0, ``choices_as_values`` option will go away, but it's behavior will
```

Comment:
Since Symfony 3.0 will be released in 4 weeks, we should not use the future tense to make this article future-proof:

```
In Symfony 3.0, ``choices_as_values`` option isn't defined, but it's behavior is always used:
```